frontend/src/scenes/surveys/SurveyFormAppearance.tsx

```diff
@@ -23,7 +24,16 @@ export function SurveyFormAppearance({
             <SurveyAppearancePreview
                 survey={survey as Survey}
                 previewPageIndex={previewPageIndex}
-                onPreviewSubmit={() => handleSetSelectedPageIndex(previewPageIndex + 1)}
+                onPreviewSubmit={(response) => {
+                    handleSetSelectedPageIndex(
+                        getNextQuestionIndex(
```

Comment:
Since the JS SDK is responsible for rendering the preview and the SDK already [knows whats the next question](https://github.com/PostHog/posthog-js/blob/4c50987442b6fb662792365604e524961ce00dd8/src/posthog-core.ts#L1315), would be possible to let the JS SDK handle this altogether?